test/solidlsp/swift/test_swift_basic.py

```diff
@@ -0,0 +1,228 @@
+        assert user_def.get("uri", "").endswith("main.swift"), "Definition should be in main.swift"
+        
+        # The User struct is defined starting at line 26
+        start_line = user_def.get("range", {}).get("start", {}).get("line")
+        assert start_line == 25, f"User struct definition should be at line 26, got {start_line + 1}"
+
+    @pytest.mark.parametrize("language_server", [Language.SWIFT], indirect=True)
+    def test_goto_definition_calculator_method(self, language_server: SolidLanguageServer) -> None:
+        """Test goto_definition on Calculator method usage."""
+        file_path = os.path.join("src", "main.swift")
+        
+        # Find the add method usage at line 6: let result = calculator.add(5, 3)
+        # Position should be at the "add" method call
+        definitions = language_server.request_definition(file_path, 5, 28)  # Position at add method call
+        assert isinstance(definitions, list), "Definitions should be a list"
+        
+        # Verify the definition points to the add method definition
+        add_def = definitions[0]
+        assert add_def.get("uri", "").endswith("main.swift"), "Definition should be in main.swift"
+        
+        # The add method is defined starting at line 17
+        start_line = add_def.get("range", {}).get("start", {}).get("line")
+        assert start_line == 16, f"add method definition should be at line 17, got {start_line + 1}"
+
+    @pytest.mark.parametrize("language_server", [Language.SWIFT], indirect=True)
+    def test_goto_definition_cross_file(self, language_server: SolidLanguageServer) -> None:
+        """Test goto_definition across files - Utils struct."""
+        utils_file = os.path.join("src", "utils.swift")
+        
+        # First, let's check if Utils is used anywhere (it might not be in this simple test)
+        # We'll test goto_definition on Utils struct itself
+        symbols = language_server.request_document_symbols(utils_file)
+        utils_symbol = next((s for s in symbols[0] if s.get("name") == "Utils"), None)
+        
+        sel_start = utils_symbol["selectionRange"]["start"]
+        definitions = language_server.request_definition(utils_file, sel_start["line"], sel_start["character"])
+        assert isinstance(definitions, list), "Definitions should be a list"
+        
+        # Should find the Utils struct definition itself
+        utils_def = definitions[0]
+        assert utils_def.get("uri", "").endswith("utils.swift"), "Definition should be in utils.swift"
+
+    @pytest.mark.parametrize("language_server", [Language.SWIFT], indirect=True)
+    def test_request_references_calculator_class(self, language_server: SolidLanguageServer) -> None:
+        # SourceKit LSP needs a short initialization period after startup
+        # before it can provide accurate reference information. This sleep
+        # prevents race conditions where references might not be available yet.
+        time.sleep(5)
```

Comment:
This shouldn't happen in the test but rather at the first call to request_references in the LS